packages/opentelemetry-node/src/instrumentation/PluginLoader.ts

```diff
@@ -78,6 +78,16 @@ export class PluginLoader {
         return this;
       }
 
+      const alreadyRequiredModules = Object.keys(require.cache);
+      const requiredModulesToHook = modulesToHook.filter(name =>
+        alreadyRequiredModules.some(cached => cached.includes(name))
+      );
+      if (requiredModulesToHook.length > 0) {
+        this.logger.warn(
+          `Some modules (${requiredModulesToHook}) were already required when their respective plugin were loaded. They might not work.`
```

Comment:
"They might not work" implies the plugin itself might not work. I would change this to be clear that it is the instrumentation that might not work.